lib/AST/FineGrainedDependencies.cpp

```diff
@@ -77,22 +77,28 @@ void SourceFileDepGraph::forEachArc(
 
 InterfaceAndImplementationPair<SourceFileDepGraphNode>
 SourceFileDepGraph::findExistingNodePairOrCreateAndAddIfNew(
-    NodeKind k, StringRef context, StringRef name,
-    Optional<std::string> fingerprint) {
+    NodeKind k, const ContextNameFingerprint &contextNameFingerprint) {
+  const std::string &context = std::get<0>(contextNameFingerprint);
+  const std::string &name = std::get<1>(contextNameFingerprint);
+  const Optional<std::string> &fingerprint =
+      std::get<2>(contextNameFingerprint);
   InterfaceAndImplementationPair<SourceFileDepGraphNode> nodePair{
       findExistingNodeOrCreateIfNew(
           DependencyKey(k, DeclAspect::interface, context, name), fingerprint,
           true /* = isProvides */),
       findExistingNodeOrCreateIfNew(
           DependencyKey(k, DeclAspect::implementation, context, name),
           fingerprint, true /* = isProvides */)};
-  // if interface changes, have to rebuild implementation
-  addArc(nodePair.getInterface(), nodePair.getImplementation());
+  // if interface changes, have to rebuild implementation.
+  // But, if an arc is added for this, then *any* change that causes
+  // a same-named interface to be dirty will dirty this implementation,
+  // even if that interface is in another file.
+  // So, make the interface->implementation arc implicit.
```

Comment:
This is an interesting design point.  Can you expand on why this is for my edification?

Reply:
Yes, I've rewritten the comment to clarify the point. Thanks.